designs/root_help_text_design.md

```diff
@@ -0,0 +1,205 @@
+==========================================
+
+What is the problem?
+--------------------
+
+Currently AWS SAM CLI's root command help text does not showcase a coherent story on how to develop a serverless application.
+
+Current state of the world as of v1.71.0
+
+```
+Usage: sam [OPTIONS] COMMAND [ARGS]...
+
+  AWS Serverless Application Model (SAM) CLI
+
+  The AWS Serverless Application Model extends AWS CloudFormation to provide a
+  simplified way of defining the Amazon API Gateway APIs, AWS Lambda
+  functions, and Amazon DynamoDB tables needed by your serverless application.
+  You can find more in-depth guide about the SAM specification here:
+  https://github.com/awslabs/serverless-application-model.
+
+Options:
+  --debug     Turn on debug logging to print debug message generated by SAM
+              CLI and display timestamps.
+  --version   Show the version and exit.
+  --info
+  -h, --help  Show this message and exit.
+
+Commands:
+  init      Init an AWS SAM application.
+  validate  Validate an AWS SAM template.
+  build     Build your Lambda function code
+  local     Run your Serverless application locally for quick development...
+  package   Package an AWS SAM application.
+  deploy    Deploy an AWS SAM application.
+  delete    Delete an AWS SAM application and the artifacts created by sam
+            deploy.
+  logs      Fetch logs for a function
+  publish   Publish a packaged AWS SAM template to the AWS Serverless
+            Application Repository.
+  traces    Fetch AWS X-Ray traces
+  sync      Sync a project to AWS
+  pipeline  Manage the continuous delivery of the application
+```
+
+What will be changed?
+---------------------
+
+The help text should showcase the stages in the lifecycle of development in a serverless application.
+
+```
+Usage: sam [OPTIONS] COMMAND [ARGS]...
+
+  AWS Serverless Application Model (SAM) CLI
+
+  The AWS Serverless Application Model extends AWS CloudFormation to provide a
+  simplified way of defining the Amazon API Gateway APIs, AWS Lambda
+  functions, and Amazon DynamoDB tables needed by your serverless application.
+  You can find more in-depth guide about the SAM specification here:
+  https://github.com/awslabs/serverless-application-model.
+
+
+Commands:
+
+  *Create an App!*
+
+      init      Init an AWS SAM application
+
+  *Develop your app!*
+
+      build     Build your Lambda Function code
+      local     Run your serverless function locally
+      validate  Validate an AWS SAM template
+
+  *Develop your app in the Cloud! (NEW)*
+      sync      Sync a project to AWS
+
+  *Deploy your app to the Cloud!*
+
+      package   Package an AWS SAM application
+      deploy    Deploy an AWS SAM application
+      delete    Delete an AWS SAM application and the artifacts created by sam
+                deploy
+      publish   Publish a packaged AWS SAM template to AWS Serverless Application Repository
+
+  *Monitor your app!*
+
+      logs      Fetch logs for a function
```

Comment:
Fetch cloud watch logs

Reply:
from the cloud is a good thought, I wonder if we should move all the AWS based acrnoyms over the top of the help text.

Reply:
we can drop AWS, but we might support more 3p (like Terraform today) which might bring in confusion